pkg/search/postgres/query/common.go

```diff
@@ -91,5 +95,26 @@ func MatchFieldQuery(dbField *walker.Field, value string, highlight bool, now ti
 	if !ok {
 		return nil, nil
 	}
-	return matchFieldQuery(dbField, field, value, highlight, now)
+
+	qualifiedColName := dbField.Schema.Table + "." + dbField.ColumnName
+	dataType := dbField.DataType
+	var goesIntoHavingClause bool
+	if derivedMetadata != nil {
+		switch derivedMetadata.DerivationType {
+		case search.CountDerivationType:
+			qualifiedColName = fmt.Sprintf("count(%s)", qualifiedColName)
+			dataType = walker.Integer
+			goesIntoHavingClause = true
+		default:
+			return nil, errors.Errorf("unsupported derivation type %s", derivedMetadata.DerivationType)
```

Comment:
Just a note that I added another derivation type after this PR was opened.

Reply:
Yeah, I saw that. I think it's going to be harder to support priorities for selects... in either case, I'll update this message to say "unsupported derivation type in select"